pkg/porter/cnab.go

```diff
@@ -1,10 +1,262 @@
 package porter
 
-import cnabprovider "github.com/deislabs/porter/pkg/cnab/provider"
+import (
+	"bufio"
+	"github.com/deislabs/porter/pkg/config"
+	"github.com/deislabs/porter/pkg/parameters"
+	"github.com/pkg/errors"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	cnabprovider "github.com/deislabs/porter/pkg/cnab/provider"
+)
 
 // CNABProvider
 type CNABProvider interface {
 	Install(arguments cnabprovider.InstallArguments) error
-	//Upgrade() error
-	//Uninstall() error
+	Uninstall(arguments cnabprovider.UninstallArguments) error
+}
+
+
+// sharedOptions are common options that apply to multiple CNAB actions.
+type sharedOptions struct {
```

Comment:
sharing is caring!

Reply:
Yeah I think we'll want to do this for the duffle Install/Uninstall args too but I wasn't confident so I held off on that for now. It seems that pretty much all the cnab actions take the same options and just treat things slightly differently. I didn't want to consolidate too quickly so I wanted to wait until we had all the actions implemented fully, then see if they are all the same.